### PR TITLE
Remove node_num function.

### PR DESCRIPTION
**Pull request by gongweibao**
`node_num` is not needed for users, so remove them and fix the bugs about it!

### CODE REVIEW

File: python/paddle/fluid/incubate/fleet/collective/__init__.py [hunk 1]
```diff
@@ -249,7 +222,67 @@ def _transpile(self, startup_program, main_program):
...
+        node_num = self._node_num()
+        assert node_num >= 1, "nccl2 node_num must >= 1, now:{}" % node_num
+
+        self._strategy.fuse_all_reduce_ops = True
+        exec_strategy = self._strategy.exec_strategy
+
+        if node_num <= 1:
+            self._strategy.nccl_comm_num = 1
+            if self._strategy.nccl_comm_num > 1:
+                logging.warn("set nccl_comm_num=1 since you only have 1 node.")
```
Comment (danleifeng) [inline python/paddle/fluid/incubate/fleet/collective/__init__.py:259]: self._strategy.nccl_comm_num = 1 may put the end of the condition.

File: python/paddle/fluid/incubate/fleet/collective/__init__.py [hunk 2]
```diff
@@ -249,7 +222,67 @@ def _transpile(self, startup_program, main_program):
...
+
+            self._strategy.use_hierarchical_allreduce = False
+            if self._strategy.use_hierarchical_allreduce:
+                logging.warn(
+                    "set use_hierarchical_allreduce=False since you only have 1 node."
+                )
```
Comment (danleifeng) [inline python/paddle/fluid/incubate/fleet/collective/__init__.py:265]: self._strategy.use_hierarchical_allreduce = False may put the end of the condition.

File: python/paddle/distributed/launch.py [hunk 3]
```diff
@@ -139,6 +139,7 @@ def start_procs(args):
 
     current_node_ip = args.node_ip
     node_ips = [x.strip() for x in args.cluster_node_ips.split(',')]
+    node_num = len(node_ips)
     node_id = node_ips.index(current_node_ip)
     num_nodes = len(node_ips)
```
Comment (danleifeng) [inline python/paddle/distributed/launch.py]: variable ```node_num```  equals to ```num_nodes```. maybe we can change all ```num_nodes``` to ```node_num``` and rm ```num_nodes```.

File: python/paddle/fluid/incubate/fleet/collective/__init__.py (hunk 1 above)
Comment (gongweibao) [reply python/paddle/fluid/incubate/fleet/collective/__init__.py:259]: Thanks! Done!

File: python/paddle/fluid/incubate/fleet/collective/__init__.py (hunk 2 above)
Comment (gongweibao) [reply python/paddle/fluid/incubate/fleet/collective/__init__.py:265]: Thanks! Done!

File: python/paddle/distributed/launch.py (hunk 3 above)
Comment (gongweibao) [reply python/paddle/distributed/launch.py]: Thanks! Done!

**Review by seiriosPlus — APPROVED** [review]
LGTM

**Pull request merged**